drgn_tools/scsiHosts.py

```diff
@@ -0,0 +1,60 @@
+# Copyright (c) 2023, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/
+"""
+Helper to print scsi hosts
+"""
+from typing import List
+
+from drgn import container_of
+from drgn import Object
+from drgn import Program
+from drgn.helpers.linux.list import list_for_each_entry
+
+
+def _list_of_scsi_host(prog: Program) -> List[Object]:
+    """
+    Iterates through all scsi hosts and returns a
+    list of scsi devices
+    """
+    try:
+        class_in_private = prog.cache["knode_class_in_device_private"]
+    except KeyError:
+        class_in_private = prog.type("struct device_private").has_member(
+            "knode_class"
+        )
+        prog.cache["knode_class_in_device_private"] = class_in_private
+
+    devices = prog["shost_class"].p.klist_devices.k_list.address_of_()
+
+    shostlist = []
+
+    if class_in_private:
+        for device_private in list_for_each_entry(
+            "struct device_private", devices, "knode_class.n_node"
+        ):
+            dev = device_private.device
+            shostlist.append(
+                container_of(dev, "struct Scsi_Host", "shost_dev")
+            )
+        return shostlist
+    else:
+        for devs in list_for_each_entry(
+            "struct device", devices, "knode_class.n_node"
+        ):
+            shostlist.append(
+                container_of(devs, "struct Scsi_Host", "shost_dev")
+            )
+        return shostlist
+
+
+def print_scsi_hosts(prog: Program) -> None:
+    """
+    Prints scsi host information
+    """
+    shostlist = _list_of_scsi_host(prog)
+    print("SCSI HOSTS\n==========\nHOST\t\t\tNAME")
+    for x in range(len(shostlist)):
+        # print(f"{shostlist[x].hostt.name}\n")
+        print(
+            f"{hex(shostlist[x].value_())}\thost{shostlist[x].host_no.value_()}"
+        )
```

Comment:
Try using the `drgn_tools.table.print_table()` function to help this. It's the standard way to print these things so they get formatted properly.